### PR TITLE
Add support to deploy local and remote files

### PR DESCRIPTION
**Pull request by aguerra**
For now only http and https are supported.

### CODE REVIEW

File: pkg/client/tar/tar.go [hunk 1]
```diff
@@ -92,3 +92,72 @@ func CreateTemp(dir, prefix string, ignorePatterns []string) (string, error) {
...
 	return tmp.Name(), nil
 }
+
+func ExtractToTemp(filename string) (string, error) {
+	gr, rc, err := newReadClosers(filename)
+	if err != nil {
+		return "", err
+	}
+	defer rc.Close()
+	defer gr.Close()
```
Comment (drgarcia1986) [inline pkg/client/tar/tar.go]: 💅 
```go
defer func() {
    rc.Close()
    gr.Close()
}()
```

File: pkg/client/cmd/deploy.go [hunk 2]
```diff
@@ -161,13 +166,34 @@ func deployApp(cmd *cobra.Command, args []string) {
...
+	if err != nil {
+		client.PrintErrorAndExit("Error accessing path: %v", err)
+	}
+	if finfo.Mode().IsRegular() {
+		path, err = tar.ExtractToTemp(path)
+		if err != nil {
+			client.PrintErrorAndExit("Error extracting archive: %v", err)
+		}
+		defer os.RemoveAll(path)
+	}
```
Comment (drgarcia1986) [inline pkg/client/cmd/deploy.go]: What do you think about to break this code snippet in a small function?

Comment (aguerra) [reply pkg/client/cmd/deploy.go]: Yeap, deploy is already too big

File: pkg/client/tar/tar.go (hunk 1 above)
Comment (aguerra) [reply pkg/client/tar/tar.go]: I guess it shoud be
```go
defer func() {
    gr.Close()
    rc.Close()
}()
```

Comment (drgarcia1986) [reply pkg/client/tar/tar.go]: you're right

**Pull request merged**